### DISCUSSION TITLE
Disabling random selection of container element for O1 strategy

### DISCUSSION THREAD

**Discussion by avolchek** [beartype/beartype#383] [Q&A]
Hi!

The documentation states that for the O1 strategy, only one element of the container is selected, and the selection is random. Is it possible to replace random selection with a more deterministic approach (e.g., selecting the first element)?

The problem with random selection is that it makes tests in CI less robust. For example, I enabled type-checking in some tests in my codebase, and they passed successfully. However, this was only because the random selection chose a "good" element from the container. The next day, my colleague changed an unrelated piece of code, altering the RNG state. As a result, a "bad" element was selected, causing Beartype to complain, and the test failed, leading to my colleague's frustration.

My codebase is large, with many people working on it. It would be great to be able to disable randomness via configuration to avoid situations like the one mentioned above :)

**Comment by leycec** [comment]
**Feature request #385,** I summon you! :mage_man:
